microsoft/community-organization-operations-suite
# 655 offline authentication

**What** 
 - Allow the user to login when the app is offline

**Why**
 - So a user can still use the app when not connected to the internet

**How**
 - The user must have previously logged into the app while online
 - To perform offline authentication we save the user's encrypted password after a successful online authentication, then when logging in offline, we encrypt their password and compare it to the saved encrypted password. We also save the access token so it can be used to send requests when app goes back online. A better solution might be obtain new token with user credentials when app goes back online. Or even have the user re-authenticate when app returns online.

**Testing**

1.  login online
2. set app to offline mode
3. logout
4. login again
5. ensure cached entities are displayed
6. create some entities
7. set app to online
8. ensure created entities are persisted.
